library/common/http/client.cc

```diff
@@ -101,6 +95,7 @@ void Client::DirectStreamCallbacks::encodeData(Buffer::Instance& data, bool end_
                                 GetStreamFilters::ALLOW_FOR_ALL_STREAMS));
   direct_stream_.saveLatestStreamIntel();
   if (end_stream) {
+    direct_stream_.saveFinalStreamIntel();
```

Comment:
how about putting these in closeStream()?

Reply:
That's was my first idea - it crashes. Sigh.